collections/_posts/2019-08-29-adding-custom-doh-resolvers-to-dnscloak.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Adding Custom DNS over HTTPS Resolvers to DNSCloak
+author: nitrohorse
+editor: Jonah
+layout: post
+tags:
+  - guides
+  - dns
+  - dnscrypt
+  - ios
+---
+
+**[DNSCloak](https://apps.apple.com/us/app/dnscloak-secure-dns-client/id1452162351)** is an [open-source](https://github.com/s-s/dnscloak) DNSCrypt and DNS over HTTPS (DoH) client for iOS, which gives users the ability to encrypt their DNS requests through the use of an on-device VPN profile.
+
+While DNSCloak is highly configurable, its user interface can be unintuitive to less tech-savvy users and doesn't easily allow users to use custom DoH resolvers, apart from the default ["public-resolvers" list](https://github.com/DNSCrypt/dnscrypt-resolvers/blob/master/v2/public-resolvers.md) that the DNSCrypt project provides.
+
+Before diving in, it's important to understand that while there is a lot of nuance to DNSCrypt and DoH, these two DNS protocols essentially achieve the same goals: They both provide users with the ability to encrypt all DNS traffic to the users' desired [upstream provider(s)](https://www.privacytools.io/providers/dns/#icanndns), while preventing [DNS hijacking](https://en.wikipedia.org/wiki/DNS_hijacking), [spoofing](https://en.wikipedia.org/wiki/DNS_spoofing), and eavesdropping by 3rd parties.
+
+The development of these DNS protocols is exciting, but unlike Android 9 which has [built-in support for DNS over TLS](https://support.google.com/android/answer/9089903) (another protocol with similar goals), iOS unfortunately does not come with any built-in encrypted DNS support. Thus, DNSCloak fills the gap for iOS users to start benefitting from these protocols today.
```

Comment:
I wonder if you should say yet or if there are any iOS feature request forums that you could link to?

Reply:
Hmm good call out, it looks like using an [MDM profile could change your DNS](https://stackoverflow.com/questions/19968042/ios-configuration-profiles-to-change-dns-settings) but I’m not sure yet if this is the same thing has [“supervised mode”](https://developer.apple.com/documentation/networkextension/dns_proxy_provider).

Reply:
>... it appears that the only way to use the new DNS Proxy Network Extension in iOS 11 is through Mobile Device Management (MDM) Tools as a supervised device. This makes it impossible to install for the average user through the App Store.

> An Apple representative confirmed this is currently only for supervised devices but he seems to indicate they are considering opening this up to all devices but with no assurances or timeline.

— https://dnsdisco.com/iOS-dns-proxy-post.html

Reply:
I could add “yet” and link to this I think.

Reply:
Updated